### PR TITLE
Ios announcement

### LINKED ISSUE #708: Announcement for iOS
## Kind (Required)

- Other

 ## Overview (Required)

- We don't have the design for iOS. 🙇 
image

### PR DESCRIPTION
**Pull request by MihoSasaki**
## Issue
- close #708 

## Overview (Required)
- This pr aims to have an announcements screen. This pr only implements view components, and hasn't yet implemented logic of api call. I will add another pr to implement this following this pr. 

## Links
-

## Implementation steps
- [x] add announcement screen
- [x] obtain data from announcement api
  - this is implemented in this pr(https://github.com/MihoSasaki/conference-app-2019/pull/1)
- [x] replace wip icon with icon same as Android 
- [ ] time format
- [ ] make text in the content of announcement clickable if it is URL or deeplink
  - this library may meet this needs...?(https://github.com/TTTAttributedLabel/TTTAttributedLabel)

### CODE REVIEW

**Comment by MihoSasaki** [comment]
I will resolve the conflict later 🐈

**Comment by takahirom** [comment]
@MihoSasaki Why WIP?

File: frontend/ios/DroidKaigi 2019/Scenes/Announcement/AnnouncementDataSource.swift [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+//}
+
+extension AnnouncementDataSource: UITableViewDelegate {
+
+    func tableView(_ tableView: UITableView, didSelectRowAt indexPath: IndexPath) {
+        tableView.deselectRow(at: indexPath, animated: false)
+    }
+    
+    func tableView(_ tableView: UITableView, willDisplay cell: UITableViewCell, forRowAt indexPath: IndexPath) {
+        cellHeightsCache[indexPath] = cell.frame.height
```
Comment (ShoMasegi) [inline frontend/ios/DroidKaigi 2019/Scenes/Announcement/AnnouncementDataSource.swift]: You don't need to cache this cell height.  
we need to cache it only when heavy calculation is required. (ex. when TableViewCell have collectionViewCell)

File: frontend/ios/DroidKaigi 2019/Scenes/Announcement/AnnouncementTableViewCell.swift [hunk 2]
```diff
@@ -0,0 +1,29 @@
+//
+//  AnnouncementTableViewCell.swift
+//  DroidKaigi 2019
+//
+//  Created by 佐々木美穂 on 2019/02/02.
+//
+
+import UIKit
+
+class AnnouncementTableViewCell: UITableViewCell, Reusable {
```
Comment (ShoMasegi) [inline frontend/ios/DroidKaigi 2019/Scenes/Announcement/AnnouncementTableViewCell.swift]: You don't need to implement `Reusable` if you don't use `UITableView+Extension`.

**Comment by MihoSasaki** [comment]
@takahirom 
I implemented announcement api logics following this pr( (https://github.com/MihoSasaki/conference-app-2019/pull/1) cuz I was afraid that this pr could be very big.
Can I merge the pr to this pr, or is it better for you to look at both prs independently?
I'm not familiar with a way of committing to OSS, so sorry if it is wrong🙏

File: frontend/ios/DroidKaigi 2019/Assets.xcassets/alert.imageset/Contents.json [hunk 3]
```diff
@@ -0,0 +1,22 @@
+{
+  "images" : [
+    {
+      "idiom" : "universal",
+      "filename" : "exclamation-mark.png",
```
Comment (MihoSasaki) [inline frontend/ios/DroidKaigi 2019/Assets.xcassets/alert.imageset/Contents.json]: I added temporary image resources. Is it possible to get resources same as Android as png images...?

File: frontend/ios/DroidKaigi 2019/Scenes/MainViewController.swift [hunk 4]
```diff
@@ -78,6 +78,9 @@ extension MainViewController: DrawerViewControllerDelegate {
         case .floorMap:
             let floorMap = FloorMapViewController.instantiateFromStoryboard()
             navigationController?.pushViewController(floorMap, animated: false)
+        case .announce:
+            let announce = AnnouncementsViewController.instantiateFromStoryboard()
+            navigationController?.pushViewController(announce, animated: false)
```
Comment (MihoSasaki) [inline frontend/ios/DroidKaigi 2019/Scenes/MainViewController.swift]: I changed my implementation following the change of moving menus to navigation drawer 🙋

Comment (MihoSasaki) [reply frontend/ios/DroidKaigi 2019/Scenes/MainViewController.swift]: I will delete unused icon image🐈

File: frontend/ios/DroidKaigi 2019/Assets.xcassets/alert.imageset/Contents.json (hunk 3 above)
Comment (jmatsu) [reply frontend/ios/DroidKaigi 2019/Assets.xcassets/alert.imageset/Contents.json]: @MihoSasaki 
Could you please use icons in https://drive.google.com/open?id=1R6lEMomFePkfWtmoCwfIfR2c7N3qc9h7 ? Also, it's okay to do it on another PR.

Comment (MihoSasaki) [reply frontend/ios/DroidKaigi 2019/Assets.xcassets/alert.imageset/Contents.json]: I applied them to both prs🐱

**Comment by MihoSasaki** [comment]
I added commits of implementing api logics(https://github.com/MihoSasaki/conference-app-2019/pull/1) because it won't be big changes 🙇‍♀️

File: frontend/ios/DroidKaigi 2019/data/ApiHandleRepository.swift [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+//
+//  Created by 佐々木美穂 on 2019/02/03.
+//
+
+import Foundation
+import ioscombined
+
+//FIXME: If you have a more better idea, we can refine this.
+func handledKotlinException(_ error: Error) -> Error {
+
```
[the remark this answers is not in the document]
Comment (MihoSasaki) [reply frontend/ios/DroidKaigi 2019/data/ApiHandleRepository.swift]: Sorry, I couldn't get what you are expecting me to do😿 
Is it not good that I move this function to this file?

[the remark this answers is not in the document]
Comment (jmatsu) [reply frontend/ios/DroidKaigi 2019/data/ApiHandleRepository.swift]: Oops, sorry. I haven't noticed this is just refactoring.

My expectation was like below. 

```
func handledKotlinException(_ error: Error) -> Error {
    // Return handled kotlin exception.
    if let handledException = cause as? KotlinException {
        return handledException
    }
    
    guard let cause = error as? KotlinThrowable else {
        fatalError("Unexpedeted Error: \(error)")
    }
    
    // Obtain `NSError` from Ktor `IosHttpRequestException`.
    // See also: data/api/ThrowableExt.kt
    if let origin = cause.originNSError {
        return origin
    }
    
    fatalError("Unexpedeted KotlinThrowable: \(cause)")
}
```

In Java/Kotlin world, Exception extends Throwable so the previous control flow sounds weird, right? And I'm not sure KotlinThrowable extends KotlinException in Kotlin/Native world. 🤷‍♂️ That's what I'd like to say through *there is no inheritance relation between KotlinThrowable and KotlinException*.

BTW, the following is the best, to be honest. You don't need to have to do refactoring anymore though.

```
func handledKotlinException(_ error: Error) -> Error {
    // Return handled kotlin exception.
    if let handledException = cause as? KotlinException {
        return handledException
    }
    
    if let cause = error as? KotlinThrowable, origin = cause.originNSError else {
        return origin
    }
    
    fatalError("Unexpedeted KotlinThrowable: \(cause)")
}
```

[the remark this answers is not in the document]
Comment (jmatsu) [reply frontend/ios/DroidKaigi 2019/data/ApiHandleRepository.swift]: Anyway, your changes to extract this method are nice.

[the remark this answers is not in the document]
Comment (MihoSasaki) [reply frontend/ios/DroidKaigi 2019/data/ApiHandleRepository.swift]:
```
if let cause = error as? KotlinThrowable, origin = cause.originNSError else {
        return origin
    }
```
swift cannot write like this, so I just changed order of the code 🐈

**Comment by MihoSasaki** [comment]
@jmatsu I fixed some issues from the comments! Could you take a look at this again?🐱

**Pull request merged**